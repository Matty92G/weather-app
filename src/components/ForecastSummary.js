import React from "react";
import iconData from "../data/iconData.json";

function ForecastSummary(props) {
  const { date, icon, temperature, description } = props;
  const formattedDate = new Date(date).toDateString();
  const weatherCode = icon.slice(0, 1) * 100;
  return (
    <div className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary_date">{formattedDate}</div>
      <div className="forecast-summary_icon" data-testid="forecast-icon">
        <img src={iconData[weatherCode]} alt={weatherCode} />
      </div>
      <div className="forecast-summary_temperature">
        {temperature.max}
        &deg;C
      </div>
      <div className="forecast-summary_description">{description}</div>
    </div>
  );
}

export default ForecastSummary;
