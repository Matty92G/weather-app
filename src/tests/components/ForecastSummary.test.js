import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 11111111,
    icon: "stubIcon",
    description: "stubDescription",
    temperature: {
      min: 10,
      max: 99,
    },
  };
  it("renders and matches snapshot", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        icon={validProps.icon}
        description={validProps.description}
        temperature={validProps.temperature.max}
      />
    );
    expect(asFragment).toMatchSnapshot();
  });
});
