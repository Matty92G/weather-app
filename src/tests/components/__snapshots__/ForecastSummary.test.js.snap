// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders and matches snapshot 1`] = `[Function]`;
